cmd/kubeadm/app/phases/controlplane/manifests.go

```diff
@@ -162,7 +162,12 @@ func getAPIServerCommand(cfg *kubeadmapi.ClusterConfiguration, localAPIEndpoint
 		}
 	} else {
 		// Default to etcd static pod on localhost
-		defaultArguments["etcd-servers"] = fmt.Sprintf("https://127.0.0.1:%d", kubeadmconstants.EtcdListenClientPort)
+		// localhost IP family should be the same that the AdvertiseAddress
+		etcdLocalhostAddress := "127.0.0.1"
+		if utilsnet.IsIPv6String(localAPIEndpoint.AdvertiseAddress) {
+			etcdLocalhostAddress = "::1"
```

Comment:
Our etcd certificate allows `localhost`, wouldn't it be better to use it instead?

Reply:
localhost is a lottery, you don't known in advance what ip is going to resolve, check the following gist explaining the topic
https://gist.github.com/aojea/94f6f483173641647c731f582e52f0b0

Reply:
Yes, it's better to use IP addresses directly. I was thinking that using `localhost` we would rely on whatever has been set on the containers `/etc/hosts`, but that's just shifting the problem somewhere else.